vespa/package.py

```diff
@@ -534,16 +537,21 @@ def __init__(
         self.output = output_file
 
     @staticmethod
-    def _read_private_key(key_location: str) -> ec.EllipticCurvePrivateKey:
-        with open(key_location, "rb") as key_data:
-            key = serialization.load_pem_private_key(
-                key_data.read(), None, default_backend()
+    def _read_private_key(
+        key_location: Optional[str] = None, key_content: Optional[str] = None
+    ) -> ec.EllipticCurvePrivateKey:
+        if not key_content:
+            with open(key_location, "rb") as key_data:
+                key_content = key_data.read()
+        else:
+            key_content = bytes(key_content, "ascii")
```

Comment:
You should check both `key_content` and `key_location` and fail with an error message if both are `None`.